grails-app/services/org/broadinstitute/orsp/ConsentService.groovy

```diff
@@ -513,6 +514,26 @@ class ConsentService implements Status {
         }
     }
 
+    /**
+     * Find all consent groups associated to the specified consent collection links
+     *
+     * @param keys The consent group key and related consent collection link
+     * @return List of Consent Group that are related to the specified consent collection links
+     */
+    Collection<Issue> findConsentGroupsAssociatedByConsentLink(Map<String, ConsentCollectionLink> keys) {
+        if (keys && !keys.isEmpty()) {
+            List<Issue> issues = Issue.findAllByProjectKeyInList(keys.keySet().toList())
+            issues.each { issue ->
+                // set all documents related to the specified consent group including its consent collection link documents
+                issue.setAttachments(queryService.getAttachmentsForProjects(keys.keySet()))
+                issue.setStatus(keys.get(issue.projectKey).status)
+            }
+            issues
+        } else {
+            Collections.emptyList()
+        }
+    }
+
     Boolean collectionDocumentsApproved(List<Long> ids){
         List<StorageDocument> documents = queryService.findAllDocumentsBySampleCollectionIdList(ids)
         ArrayList<StorageDocument> pendingDocuments = documents.findAll {
```

Comment:
Technically not related to this PR, but I noticed this while reviewing. This block is trying to coerce a list of strings to a list of `StorageDocuments`:
```
        ArrayList<StorageDocument> pendingDocuments = documents.findAll {
            it.status == DocumentStatus.PENDING.status
        }.fileType
```
Here's one way we could avoid the multiple mutated lists with some the spread operator (*), some chained streaming and null-safe operators:
```
        queryService.
                findAllDocumentsBySampleCollectionIdList(ids)*.
                status?.
                findAll { it == DocumentStatus.PENDING.status}?.
                size() == 0
```
There are probably better ways to answer the question, but this is one.